File: An2/MAP/MAP-Java/MAP-Lab8/src/Model/Student.java
package Model;

/**
 * Created with IntelliJ IDEA.
 * User: mihai
 * Date: 12/4/13
 * Time: 11:44 PM
 * To change this template use File | Settings | File Templates.
 */
public class Student implements Comparable<Student>, HasID {

    private int id;
    private String name;
    private int grade;

    public Student (int id, String name, int grade) {
        this.name = name;
        this.id = id;
        this.grade = grade;
    }

    public float average(){
        return grade;
    }

    public int getID() {
        return this.id;
    }

    public int getGrade() {
        return this.grade;
    }

    public String getName(){
        return this.name;
    }

    @Override
    public String toString() {
        return id + " " + name + " " + grade;
    }

    @Override
    public boolean isGreaterThan(Student student) {
        return grade > student.grade;
    }

}

File: An2/Sem1/MAP/MAP-C#/MAP-Lab10/MAP-Lab10/Model/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_Lab10.Model
{
    class Student : Comparable<Student>, HasID
    {
        private string name;
        private int id;
        private int grade;

        public Student()
        {

        }

        public Student(int id, string name, int grade)
        {
            this.id = id;
            this.name = name;
            this.grade = grade;
        }

        public int getID()
        {
            return this.id;
        }

        public int getGrade()
        {
            return this.grade;
        }

        public String getName()
        {
            return this.name;
        }

        public virtual float average()
        {
            return grade;
        }

        bool Comparable<Student>.isGreaterThan(Student student)
        {
            return (this.grade > student.grade);
        }

        public override String ToString()
        {
            return this.id + " " + this.name + " " + this.grade;
        }
    }
}
